lightwood/analysis/nc/norm.py

```diff
@@ -1,24 +1,110 @@
 import math
-import torch
+from typing import Union
 
+import torch
+import numpy as np
+import pandas as pd
+from scipy.stats import entropy
+from scipy.special import softmax
+from sklearn.linear_model import Ridge
+from sklearn.metrics import mean_absolute_error
+
+from lightwood.api.dtype import dtype
+from lightwood.analysis.nc.nc import BaseScorer
 from lightwood.helpers.device import get_devices
 from lightwood.helpers.torch import LightwoodAutocast
-from lightwood.analysis.nc.nc import BaseScorer
+from lightwood.data.encoded_ds import ConcatedEncodedDs
+
+
+class SelfawareNormalizer(BaseScorer):
```

Comment:
Any reason we ought not to have these normalizers inherit from `BaseModel` instead?

Reply:
I.e. that has the added advantage we could easily get existing models to behave as normalizer and vice/versa.

Reply:
Nope, `BaseScorer` is fairly minimal. I've changed to inherit from `BaseModel` and removed `SelfAwareNet` (basically a copy of `DefaultNet`, ergo redundant now).

Will further tinker with this as part of the `nc` interface simplification effort, so that e.g. `score()` isn't needed anymore.